java/src/main/java/com/google/security/binexport/BinExport2Builder.java

```diff
@@ -411,20 +411,21 @@ private void buildCallGraph() throws CancelledException {
     }
   }
 
-  private void buildDataReferences(Map<Long, Integer> instructionIndices ) {
-	  monitor.setMessage("Exporting data references");
-	  monitor.setIndeterminate(true);
-
-	  for (Map.Entry<Long, Integer> insnIndex: instructionIndices.entrySet()) {
-		  Address addr = getAddressFromLong(insnIndex.getKey());
-		  for (Reference ref: program.getReferenceManager().getReferencesFrom(addr)) {
-			  if (ref.isMemoryReference() && ref.getReferenceType().isData()) {
-				  builder.addDataReferenceBuilder()
-				  .setInstructionIndex(insnIndex.getValue())
-				  .setAddress(ref.getToAddress().getOffset());
-			  }
-		  }
-	  }
+  private void buildDataReferences(Map<Long, Integer> instructionIndices) {
+    monitor.setMessage("Exporting data references");
+    monitor.setIndeterminate(true);
```

Comment:
We can do better than indeterminate progress: Set the maximum to the number of instruction indices and increase progress at the beginning of the loop.